spec/requests/image_uploads_spec.rb

```diff
@@ -76,7 +76,7 @@
       before do
         sign_in user
         allow(Rails).to receive(:cache).and_return(memory_store)
-        Rails.cache.clear
+        RedisRailsCache.clear
```

Comment:
I think the issue you're having in the tests is due to the fact that here you're still mocking `Rails.cache` and not `RedisRailsCache`.

The code calls `RedisRailsCache.read/write`, so the mocking should be done on that class